### DISCUSSION TITLE
Inventree on Docker is not compatible with postgres > 13

### DISCUSSION THREAD

**Discussion by miggland** [inventree/InvenTree#4116] [General]
Hi,

originally, this was a question. I have now found the answer, but thought I'd post it here in case anyone else is considering upgrading postgresql to major version 14 or 15. This is not covered explicitly in the documentation - but the version of postgres is fixed to 13 in the docker-compose file. I don't see any issues with this, it's just for information.

The current Docker image, being based on Debian Buster, via `python:3.9-slim`, does not have a `postgres-client` package compatible with anything above postgres 13. Debian Bullseye does not either, it seems, [version 15 is in Debian Bookworm](https://packages.debian.org/bookworm/postgresql-client-15), which is not released yet.

Of course, there are ways to install the binaries for postgres anyway, but that would require adding the corresponding repositories.

In short: Inventree will be locked to postgres 13 until Debian bookworm is released.

**Comment by miggland** [comment]
...one of the reasons being `pg_dump` which is called during `invoke update`, and fails because of a version mismatch if the postgres server is version 14.

**Comment by SchrodingersGat** [comment]
@miggland thanks for the info. Do you have a specific reason for wanting to upgrade to a newer version of postgres?

**Reply by miggland** [reply]
Yes, but unrelated to Inventree: Another project I'm working on is on PG 14. I figured I'd try to keep the same version on both.

Am I corrent that this is only caused by `pg_dump` (called by Django's databasebackup) and the database backup step? Django states it is compatible with PG 14, and might not use the postgres-client on the operating system? The client itself hasn't been in the Docker image for long (see https://github.com/inventree/InvenTree/pull/3802).

**Reply by SchrodingersGat** [reply]
Correct, it looks like the incompatibilities arise due to the requirement for `pg_dump`
